### ISSUE TITLE
Add ESM support

### ISSUE THREAD

**Issue by ahmadbilaldev** [alallier/reload#302]
Change the module from CommonJS to ECMAScript Module (ESM). Some JS Frameworks support ESM now and importing CommonJS modules may get complicated. Given that Node.js 12 has become a MAINTENANCE LTS, ECMAScript modules are the official standard format to package JavaScript code for reuse. Converting to ESM is recommended because ESM can still import CommonJS packages, but CommonJS packages cannot import ESM packages synchronously.

Two possible solutions are:

1. Completely move to JavaScript ES Modules.
2. Introduce a build step that transpiles a CommonJS fallback.

I can send in a PR if you agree.

**Comment by alallier** [comment]
Yeah we are overdue at looking at this. Going to leave it open we should work on this soon.

@kethinov/@lannonbr would love your opinions on what we should do
